amespahdbpythonsuite/amespahdb.py

```diff
@@ -0,0 +1,707 @@
+                            if (tokens[next]['type'] == 'TRANSFER'):
+                                parsed += tokens[current]['translation']
+                            elif (tokens[next]['type'] == 'IDENTITY'
+                                  or tokens[next]['type'] == 'NUMERIC'
+                                  or tokens[next]['type'] == 'FORMULA'
+                                  or tokens[next]['type'] == 'CHARGE'):
+                                parsed += ' ' + tokens[current]['translation']
+                            else:
+                                tokens[current].valid = False
+
+            elif tokens[current]['type'] == 'COMPARISON':
+                if prev is not None:
+                    if tokens[prev]['type'] == 'IDENTITY' and tokens[prev]['valid']:
+                        if next is not None:
+                            if tokens[next]['type'] == 'NUMERIC':
+                                parsed += ' ' + tokens[current]['translation']
+                            else:
+                                tokens[current]['valid'] = False
+
+            elif tokens[current]['type'] == 'CHARGE':
+                if prev is not None:
+                    if not (tokens[prev]['type'] == 'LOGICAL' and tokens[prev]['valid']):
+                        parsed += ' and '
+
+                parsed += ' ' + tokens[current]['translation']
+
+            elif tokens[current]['type'] == 'TRANSFER':
+                parsed += tokens[current]['translation']
+
+            elif tokens[current]['type'] == 'NAME':
+                if prev is not None:
+                    if not (tokens[prev]['type'] == 'LOGICAL' and tokens[prev]['valid']):
+                        parsed += ' and '
+
+                # THIS IS NOT CORRECTLY IMPLEMENTED
+                # parsed += ' pahdb.data.comments.str, "' + tokens[current]['translation'])'
+
+            elif tokens[current]['type'] == 'IGNORE':
+                print(f"{tokens[current].word} NOT UNDERSTOOD")
+
+                return ''
+
+            prev = current
+
+            current = next
+
+            if next is not None:
+                if next == ntokens - 1:
+                    next = None
+                else:
+                    next += 1
+
+        return parsed
+
+    def getversion(self):
+        """
+        Method to retrieve the PAHdb version.
+
+        Returns
+        -------
+        String of PAHdb version.
+
+        """
+        return self.__data['version']
+
+    def gettype(self):
+        """
+        Method to retrieve the PAHdb type.
+
+        Returns
+        -------
+        String of PAHdb type.
+
+        """
+        return self.__data['type']
+
+    def getdata(self):
+        """
+        Method to retrieve the database.
+
+        Returns
+        -------
+        Dictionary containing the parsed database.
+
+        """
+        return self.__data
+
+    @staticmethod
+    def message(text, space=55):
```

Comment:
This very helpful method will also be used in some of the other classes, perhaps this file is not the place to define it.

Reply:
This method can be called by other modules by simply importing it as `from amespahdbpythonsuite.amespahdb import message`. In theory, this method would not fit in the context of the other modules as well, and so introducing it here in the main module makes some sense. Let me know if you would prefer to include it in another place.